### PR TITLE
[Merged by Bors] - feat: generalize Algebra.TensorProduct.rightAlgebra

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
Co-authored-by: Kevin Buzzard <k.buzzard@imperial.ac.uk>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/TensorProduct/Basic.lean [hunk 1]
```diff
@@ -576,12 +576,23 @@ instance instCommRing : CommRing (A ⊗[R] B) :=
...
 
+variable [CommSemiring R]
+variable [Semiring A] [Algebra R A]
+variable [CommSemiring B] [Algebra R B]
+
 /-- `S ⊗[R] T` has a `T`-algebra structure. This is not a global instance or else the action of
 `S` on `S ⊗[R] S` would be ambiguous. -/
 abbrev rightAlgebra : Algebra B (A ⊗[R] B) :=
-  (Algebra.TensorProduct.includeRight.toRingHom : B →+* A ⊗[R] B).toAlgebra
+  includeRight.toRingHom.toAlgebra' <| fun a b => by
```
Comment (eric-wieser) [inline Mathlib/RingTheory/TensorProduct/Basic.lean:590]:
```suggestion
  includeRight.toRingHom.toAlgebra' fun a b => by
```

Comment (eric-wieser) [reply Mathlib/RingTheory/TensorProduct/Basic.lean:590]: Also, `b x` would probably be clearer than `a b`, since here `a` has type `B`.

**Review by eric-wieser — APPROVED** [review]
A slightly slicker proof:
```lean
  includeRight.toRingHom.toAlgebra' fun b x => by
    suffices LinearMap.mulLeft R (includeRight b) = LinearMap.mulRight R (includeRight b) from
      congr($this x)
    ext xa xb
    simp [mul_comm]
```

bors d+

**Comment by Ruben-VandeVelde** [comment]
bors merge

**Pull request closed without merging**